src/remote_router.rs

```diff
@@ -107,6 +115,24 @@ impl RemoteRouter {
         })
         .right_future()
     }
+
+    fn handler(&mut self, ctx: &mut <Self as Actor>::Context) -> LocalRouterHandler {
+        let (tx, rx) = oneshot::channel();
+
+        rx.into_actor(self)
+            .map(|_, this, ctx| {
+                this.connection.as_ref().map(|c| {
+                    c.connected().not().then(|| log::warn!("connection lost"));
+                });
+                // restarts the actor
+                ctx.stop();
```

Comment:
this probably needs some delay before restart, because it causes market test suite to hang using 100% CPU trying to reconnect in the infinite loop. I've used workaround, but it is dirty
https://github.com/golemfactory/yagna/pull/1237/commits/1a1239a3982a720b1a471dd093307e4922519bea

Reply:
Reesolved by #23